test/finders/songs_finder_test.rb

```diff
@@ -34,6 +35,11 @@ class SongsFinderTest < ActionController::TestCase
   end
 
   test 'retrieve all returns many songs' do
-    assert(SongsFinder.retrieve_all.count > 7) 
+    assert(SongsFinder.retrieve_all.count > 7)
+  end
+
+  test 'should partially match' do
+    assert_includes(partial_word_results, songs(:hands_to_the_heaven))
+    assert_includes(partial_word_results, songs(:glorious_day))
   end
```

Comment:
Not sure what you mean. This test is already doing that. If you're wondering why records still show after you've typed a relevant search + gibberish at the end, it's because you've required that at LEAST one word in the phrase matches (OR condition).

So even if you type gibberish at the end of a relevant search, as long as the previous words matched results will still show.